L1Trigger/L1TGEM/plugins/GEMPadDigiProducer.cc

```diff
@@ -97,20 +99,55 @@ void GEMPadDigiProducer::buildPads(const GEMDigiCollection& det_digis, GEMPadDig
     if (use16GE21_ and p->id().station() == 2)
       continue;
 
-    // set of <pad, bx> pairs, sorted first by pad then by bx
-    std::set<std::pair<int, int> > proto_pads;
+    // check that ME0 has 8-eta partitions
+    if (geometry_->stations().size() == 3) {
+      if (geometry_->station(1, 0)->superChamber(1)->chamber(1)->nEtaPartitions() !=
+          GEMPadDigi::NumberPartitions::ME0) {
+        edm::LogError("GEMPadDigiProducer") << "ME0 geometry appears corrupted";
+      }
+    }
+
+    // check that GE1/1 has 8-eta partitions
+    if (geometry_->station(1, 1)->superChamber(1)->chamber(1)->nEtaPartitions() != GEMPadDigi::NumberPartitions::GE11) {
+      edm::LogError("GEMPadDigiProducer") << "GE1/1 geometry appears corrupted";
+    }
+
+    // check that GE2/1 has 8-eta partitions
+    if (geometry_->stations().size() == 2) {
+      if (geometry_->station(1, 2)->superChamber(1)->chamber(1)->nEtaPartitions() !=
+          GEMPadDigi::NumberPartitions::GE21) {
+        edm::LogError("GEMPadDigiProducer") << "GE2/1 geometry (8 partition) appears corrupted";
+      }
+    }
+
+    // set of <pad, bx, part> pairs, sorted first by pad then by bx
+    std::set<std::tuple<int, int, int> > proto_pads;
 
     // walk over digis in this partition,
     // and stuff them into a set of unique pads (equivalent of OR operation)
     auto digis = det_digis.get(p->id());
     for (auto d = digis.first; d != digis.second; ++d) {
-      int pad_num = static_cast<int>(p->padOfStrip(d->strip()));
-      proto_pads.emplace(pad_num, d->bx());
+      unsigned pad_num = static_cast<int>(p->padOfStrip(d->strip()));
+      unsigned nPart = GEMPadDigi::NumberPartitions::GE11;
+      if (p->id().station() == 0) {
```

Comment:
can these station comparisons use `GEMSubDetId::Station` values instead of hardcoded values?